### PR TITLE
Pool size config documentation

### PR DESCRIPTION
**Pull request by PragTob**
Implements #395 :)

Not sure if this is the best way, but I added the information I wish I'd seen yesterday and also implements a small pointer from the README. Keep me honest on that, esp. my meta note on configuration :)

Thanks! :green_heart:

### CODE REVIEW

**Comment by PragTob** [comment]
@wojtekmach anything I can do to help move this along?

File: README.md [hunk 1]
```diff
@@ -208,6 +208,12 @@ Req.get!(req, url: "https://api.github.com/user").body["login"]
 
 See [`Req.Request`] module documentation for more information on low-level API, request struct, and developing plugins.
 
+## Configuration
+
+Req aims to work without global configuration to promote explicitness.
+However, you might need to adapt the HTTP pools used.
+Refer to the [`run_finch`] documenation for more information on this.
```
Comment (MrYawe) [inline README.md:215]:
```suggestion
Refer to the [`run_finch`] documentation for more information on this.
```

File: lib/req/steps.ex [hunk 2]
```diff
@@ -714,6 +714,24 @@ defmodule Req.Steps do
...
 
+  More commonly you'd add the the custom Finch pool as part of your supervision tree in your
+  `application.ex`:
+
+      children = [
+        {Finch,
+        name: MyFinch,
+        pools: %{
+          :default => [size: 70]
+        }}
```
Comment (wojtekmach) [inline lib/req/steps.ex:725]:
```suggestion
        {Finch,
         name: MyFinch,
         pools: %{
           default: [size: 70]
         }}
```

File: README.md [hunk 3]
```diff
@@ -202,6 +202,12 @@ Req.get!(req, url: "https://api.github.com/user").body["login"]
 
 See [`Req.Request`] module documentation for more information on low-level API, request struct, and developing plugins.
 
+## Configuration
+
+Req aims to work without global configuration to promote explicitness.
+However, you might need to adapt the HTTP pools used.
+Refer to the [`run_finch`] documentation for more information on this.
```
Comment (wojtekmach) [inline README.md:209]: WDYT about the following?

```suggestion
Req supports many configuration options, see [`Req.new/1`] for a full list and see each step for
more details. In particular, if you are looking for slightly lower level HTTP options like for
example timeouts, pool sizes, and certificates, see [`run_finch`] documentation.
```

**Comment by PragTob** [comment]
@wojtekmach looking good! I reworded it slightly in a588da2 as `like for example` didn't read nicely to me there :)

**Comment by wojtekmach** [comment]
Thank you!

**Pull request merged**